crates/bevy_transform/src/components/transform.rs

```diff
@@ -1,167 +1,132 @@
-use bevy_math::{Mat3, Mat4, Quat, Vec3, Vec4};
+use bevy_math::{Mat3, Mat4, Quat, Vec3};
 use bevy_property::Properties;
 use std::fmt;
 
 #[derive(Debug, PartialEq, Clone, Copy, Properties)]
 pub struct Transform {
-    value: Mat4,
+    pub translation: Vec3,
+    pub rotation: Quat,
+    pub scale: Vec3,
 }
 
 impl Transform {
     #[inline(always)]
-    pub fn new(value: Mat4) -> Self {
-        Transform { value }
+    pub fn identity() -> Self {
+        Transform {
+            translation: Vec3::zero(),
+            rotation: Quat::identity(),
+            scale: Vec3::one(),
+        }
     }
 
-    #[inline(always)]
-    pub fn identity() -> Self {
+    pub fn from_matrix(matrix: Mat4) -> Self {
+        let (scale, rotation, translation) = matrix.to_scale_rotation_translation();
+
         Transform {
-            value: Mat4::identity(),
+            translation,
+            rotation,
+            scale,
         }
     }
 
     pub fn from_translation(translation: Vec3) -> Self {
-        Transform::new(Mat4::from_translation(translation))
+        Transform {
+            translation,
+            ..Default::default()
+        }
     }
 
     pub fn from_rotation(rotation: Quat) -> Self {
-        Transform::new(Mat4::from_quat(rotation))
+        Transform {
+            rotation,
+            ..Default::default()
+        }
     }
 
     pub fn from_scale(scale: f32) -> Self {
-        Transform::new(Mat4::from_scale(Vec3::splat(scale)))
+        Transform {
+            scale: Vec3::one() * scale,
```

Comment:
You could do Vec3::splat(scale) here so it's just assignment with no multiplication.